### ISSUE TITLE
AbstractNode#getRole() can return Opal.nil, it should return undefined

### ISSUE THREAD

**Issue by Mogztter** [asciidoctor/asciidoctor.js#1066]
As the title says!

**Comment by bitPogo** [comment]
Is that only a TS thing or must a potential patch touch also some logic?

**Comment by Mogztter** [comment]
The implementation must be updated:

https://github.com/asciidoctor/asciidoctor.js/blob/cb0db00597ae7ecfd65cc717f3e67f8c284d5abe/packages/core/src/asciidoctor-core-api.js#L1037-L1039

It should be:

```js
AbstractNode.prototype.getRole = function () { 
  const role = this.$role()
  return role === Opal.nil ? undefined : role
}
```

The related tests and the type definition should also be updated accordingly.

**Comment by bitPogo** [comment]
Got it!

**Comment by bitPogo** [comment]
Does that also applies for [`getRoles`](https://github.com/asciidoctor/asciidoctor.js/blob/0440f5cb5edb7b69d13f61fd808c6ff731b7c710/packages/core/src/asciidoctor-core-api.js#L1059)?

**Comment by Mogztter** [comment]
No, the underlying method `roles` always returns an Array:

https://github.com/asciidoctor/asciidoctor/blob/6ef1cabdc699bce7767024729d133887b488e40f/lib/asciidoctor/abstract_node.rb#L195

**Comment by Mogztter** [comment]
Fixed in #1127
